### PR TITLE
Add Contentful objects

### LINKED ISSUE #9: Storage/DB solution for site data
- [x] Store GeoJSON data & building models
- [ ] Accommodate site content
- [ ] Consider CRUD updates by CJ
- [ ] FaunaDB vs. Firestore vs. something else

### PR DESCRIPTION
**Pull request by replygirl**
[Demo page](https://world-of-jackson-git-feature-contentful-utils.blacksocialists.now.sh/contentful-demo)

- [x] Add a generic `ContentfulObject` class abstracting away Contentful boilerplate
- [x] Add generated declarations for Contentful data: `yarn contentful-typescript-codegen`
- [x] Add importable `ContentfulObject` instances for each class in our data model

Closes #9 

Issues resulting from this PR:
- [ ] #33 Document schema workflow (self-assigned)
- [ ] #34 Document data model (self-assigned)
- [ ] #35 Update data model (self-assigned, **open to collaborators**)
- [ ] #36 Include non-cyclical fields in `ContentfulObject`'s `findRelated` step **(help wanted)**
- [ ] #38 Show Buildings from Contentful **(help wanted)**
- [ ] #39 Filter visible items using Map object from Contentful **(help wanted)**

### CODE REVIEW

File: types/db/contentful.d.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  address?: { lat: number; lon: number } | undefined;
+
+  /** Model */
+  model?: Asset | undefined;
+
+  /** Image */
+  image?: Asset | undefined;
+
+  /** Body */
+  body?: Document | undefined;
```
Comment (melnyczuk) [inline types/db/contentful.d.ts:26]: you don't actually need to put `| undefined`, the `?:` takes care of that.

**Comment by odbol** [comment]
This is great, thanks Imogen! 

One thing I noticed with the Demo page is that the JSON seems to have a lot of duplicate objects (e.g. every child object also contains the full JSON for the parent object containing it.) Is there a way we can reduce that duplication to save on bandwidth and RAM?

Maybe one way is to only include an id pointing to the parent object instead of the entire parent object?

**Comment by odbol** [comment]
I realize now that the JSON is only representing the in memory objects, which I assume have references to the parent objects instead of duplicate objects. That seems fine to me if that's the case; please ignore my earlier comment.

**Comment by replygirl** [comment]
#### 📝Changed routes:


| [**/contentful-demo**](https://world-of-jackson-git-feature-contentful-utils.blacksocialists.now.sh/contentful-demo) |
|:-:|
|<a href="https://world-of-jackson-git-feature-contentful-utils.blacksocialists.now.sh/contentful-demo">Screenshot of /contentful-demo</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/riNnjBcIQb_e__Kbz6X6d_HMxI4_.png">(view full size)</a> |







Commit 75612e643af8c0d2246eaa68ecce704d172642ba (https://world-of-jackson-git-feature-contentful-utils.blacksocialists.now.sh).

**Review by GrigoryGraborenko — APPROVED** [review]

**Review by sorcerix — COMMENTED** [review]
retroactively approved XD

**Pull request merged**